### PR TITLE
ENH: tractogram add dps

### PR DESCRIPTION
**Pull request by AntoineTheb**
# Quick description

New script to add data per streamline to a tractogram. Can be colors, streamline ids, sift2 weights, bundle names; your imagination is the only limit.

...

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Data

https://drive.google.com/file/d/1lagozFxH6k4U8TvaX49e6_sPWj3CptyG/view?usp=sharing

# Checklist

- [X] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [X] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scripts/scil_tractogram_add_dps.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    # Load tractogram
+    sft = load_tractogram_with_reference(parser, args, args.in_tractogram)
+
+    # Make sure the user is not unwillingly overwritting dps
+    if (args.dps_key in sft.get_data_per_streamline_keys() and
+       not args.overwrite):
+        parser.error('"{}" already in data per streamline. Use -f to force '
+                     'overwriting.'.format(args.dps_key))
+
+    # Load data and remove extraneous dimmensions
```
Comment (karanphil) [inline scripts/scil_tractogram_add_dps.py:66]: dimmensions -> dimensions

**Comment by karanphil** [comment]
This looks good! I will test it with real data soon and approve it.

File: scripts/scil_tractogram_add_dps.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+def _build_arg_parser():
+
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_tractogram',
+                   help='Input tractogram (.trk or .tck).')
+    p.add_argument('dps_file',
```
Comment (arnaudbore) [inline scripts/scil_tractogram_add_dps.py:33]: put `in_dps_file`

File: scripts/scil_tractogram_add_dps.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                   help='File containing the data to add to streamlines.')
+    p.add_argument('dps_key',
+                   help='Where to store the data in the tractogram.')
+    p.add_argument('out_tractogram',
+                   help='Output tractogram (.trk or .tck).')
```
Comment (arnaudbore) [inline scripts/scil_tractogram_add_dps.py:38]: Ask for trk file

File: scripts/scil_tractogram_add_dps.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+    logging.getLogger().setLevel(logging.getLevelName(args.verbose))
+
+    # I/O assertions
+    assert_inputs_exist(parser, [args.in_tractogram, args.dps_file],
+                        args.reference)
+    assert_outputs_exist(parser, args, args.out_tractogram)
+
```
Comment (arnaudbore) [inline scripts/scil_tractogram_add_dps.py:56]: assert output is trk otherwise either you choose to raise an error or just a warning, your call.

**Review by arnaudbore — APPROVED** [review]
Two small modifications otherwise LGTM

**Review by arnaudbore — APPROVED** [review]

**Review by karanphil — APPROVED** [review]

**Pull request merged**